pages/ai-ecosystem/graph-rag/knowledge-graph-creation.mdx

```diff
@@ -0,0 +1,152 @@
+
+### Why data modeling is crucial for GraphRAG?
+
+Here’s why it matters:
+
+- **Efficient retrieval**: Properly defined entities (nodes), relationships (edges), and properties enable accurate and efficient queries. The graph structure must reflect the actual domain knowledge, avoiding redundancy or ambiguity.
+- **Providing context for LLMs**: A well-modeled graph encodes the right context, allowing LLMs to generate meaningful and relevant responses.
+- **Supporting multi-hop reasoning**: Seamless traversal across multiple relationships is only possible with a structured graph design. Inefficient traversals can slow down query performance.
+- **More flexible iteration**: A solid starting point allows for testing, refining, and improving retrieval strategies without reworking the entire graph.
+
+Investing time in data modeling is essential for building a scalable, accurate and impactful GraphRAG system.
+
+<Cards>
+  <Cards.Card
+    title="Read more about data modeling"
+    href="/data-modeling"
+  />
+    <Cards.Card
+    title="Data modeling guide"
+    href="https://memgraph.com/blog"
+  />
+</Cards>
+
+{/* TODO: add link to Jure’s Data Modeling Guide for Dummies once available.*/}
+
+### Structured data
+
+Having data already structured is the **recommended approach** when creating a GraphRAG system. When dealing with **structured data** (e.g., CSV, JSON, Cypherl), you have the most control over how your knowledge graph is modeled. This control ensures high accuracy and relevance in your GraphRAG system.
+
+Structured data allows you to define nodes, edges, and properties explicitly, eliminating ambiguities that can arise when relying on automated tools or LLMs.
+
+Steps for structured data:
+
+1. Map entities and relationships
+2. Define properties. This includes adding metadata to nodes and edges for richer context.
+3. Import into Memgraph.
+
+<Cards>
+  <Cards.Card
+    title="Read more about data migration"
+    href="/data-migration"
+  />
+    <Cards.Card
+    title="Import best practices"
+    href="/data-migration/best-practices"
+  />
+</Cards>
+
+### Unstructured data
+
+Working with **unstructured data** (e.g. text, images or logs) requires additional preprocessing to create a valid knowledge graph. The process involves balancing automation and human oversight.
+
+#### Simplest approach
+
+Transforming unstructured data like text, images or logs into a structured knowledge graph requires additional steps. Tools like **LangChain** and **LlamaIndex** simplify this process by bridging the gap between unstructured data and graph databases like Memgraph.
+
+<Cards>
+  <Cards.Card
+    title="LlamaIndex"
+    href="/ai-ecosystem/integrations#llamaindex"
+  />
+    <Cards.Card
+    title="LangChain"
+    href="/ai-ecosystem/integrations#langchain"
+  />
+</Cards>
```

Comment:
I would link to LlamaIndex and LangChain from the text here, I wouldn't additionally emphasize it in the cards